File: src/main/java/de/blazemcworld/fireflow/util/PlayerExitInstanceEvent.java
package de.blazemcworld.fireflow.util;

import net.minestom.server.entity.Player;
import net.minestom.server.event.trait.PlayerInstanceEvent;
import org.jetbrains.annotations.NotNull;

public class PlayerExitInstanceEvent implements PlayerInstanceEvent {

    private final Player player;

    public PlayerExitInstanceEvent(Player player) {
        this.player = player;
    }

    @Override
    public @NotNull Player getPlayer() {
        return player;
    }
}

File: src/main/kotlin/de/blazemcworld/fireflow/util/PlayerExitInstanceEvent.kt
package de.blazemcworld.fireflow.util

import net.minestom.server.entity.Player
import net.minestom.server.event.trait.InstanceEvent
import net.minestom.server.event.trait.PlayerEvent
import net.minestom.server.instance.Instance

data class PlayerExitInstanceEvent(private val thePlayer: Player, private val theInstance: Instance) : InstanceEvent, PlayerEvent {
    override fun getInstance() = theInstance
    override fun getPlayer() = thePlayer
}